packages/components/src/DateTimePickers/DateTime/datetime-extraction.js

```diff
@@ -104,11 +123,13 @@ function extractPartsFromDateTime(datetime, options) {
 		return {
 			date: undefined,
 			time: undefined,
+			datetime: undefined,
 		};
 	}
 	return {
 		date: extractDateOnly(datetime, options),
 		time: extractTimeOnly(datetime, options),
+		datetime,
```

Comment:
I put datetime in state, so when props.value changed later we can do a quick check and not re-render when datetime is not changed.